php/ext/google/protobuf/protobuf.h

```diff
@@ -42,6 +42,10 @@
 #define MAX_LENGTH_OF_INT64 20
 #define SIZEOF_INT64 8
 
+/* From Chromium. */
+#define ARRAY_SIZE(x) \
+    ((sizeof(x)/sizeof(0[x])) / ((size_t)(!(sizeof(x) % sizeof(0[x])))))
```

Comment:
Why ```ARRAY_SIZE``` is like this? Is it related to implementation detail of php runtime?

Reply:
I've taken that macro from here: https://github.com/google/protobuf/blob/3beb19945b599d448c2524666effe1acdf29165f/php/ext/google/protobuf/upb.c#L8107-L8109

It's a classical macro to determine the size of constant-sized arrays in C.
For instance if you have a `char foo[25]` then `ARRAY_SIZE(foo)` will evaluate to `25`.

You can find an explanation here: https://stackoverflow.com/a/4415646

Since such a macro is much safer than hardcoding the size of arrays I've started replacing these hardcoded numbers in any PHP-extension code I touched, but I'd recommend using that macro in other places as well as soon as possible. It will probably reduce accidential mistakes in future PRs. 🙂